### PR TITLE
Release 0.5.0-2

### LINKED ISSUE #25: bpf_tc_opts should use the MaybeUninit method of creating a default
Similar to bpf_tc_hook, struct bpf_tc_opts needs to be fully zeroed or the BPF verifier will complain "libbpf: bpf_tc_opts has non-zero extra bytes"

client code can do this work, but it would be nice if the libbpf-sys library did it for the clients

Question: How does rust-bindgen determine that bpf_tc_hooks needs a MaybeUninit zero method, while bpf_tc_opts does not?

### PR DESCRIPTION
**Pull request by alexforster**
Add explicit padding members to generated structs (fixes #25)

> For some reason, libbpf rejects user-provided structs if their padding isn't zeroed out.
>
> See: https://github.com/libbpf/libbpf/blob/07ba0eeb8e0a4ed492464af55b0ca89a40161615/src/libbpf_internal.h#L241
>
> Note: we're also installing bindgen from source now because it contains a fix for rust-lang/rust-bindgen#2108. Once the next version of bindgen is released, we can use it instead.

Remove rustfmt.toml and use Ubuntu LTS to generate bindings

> * My preferred `rustfmt.toml` requires nightly features.
> * Ubuntu 20.10 is already EOL. Use 20.04 instead, which is perfectly fine our purposes.

Release 0.5.0-2

### CODE REVIEW

**Comment by alexforster** [comment]
Will this achieve what you need, @mimullin-bbry?

**Comment by mimullin-bbry** [comment]
Visual inspection looks promising.  I shall test by EOD Nov 1, 2021.

**Comment by mimullin-bbry** [comment]
@alexforster This achieves what I need Alex.  Thank you.

Warning for your downstream clients.  The addition of all these new __bindgen_padding_x fields might cause a bit of a headache.  libbpf-rs requires just two lines of change in https://github.com/libbpf/libbpf-rs/blob/master/libbpf-rs/src/object.rs#L62.  I am unsure of who your other customers are.

I have tested via local updates to the above libbpf-rs object.rs file and compiling my example application found in this libbpf-rs PR: https://github.com/libbpf/libbpf-rs/pull/138

Lines https://github.com/mimullin-bbry/libbpf-rs/blob/adding-tc/libbpf-rs/src/tc.rs#L41 have removed the MaybeUninit method of zeroing and instead are just using a ::default() call on bpf_tc_opts (similar to bpf_tc_hook).

TL;DR +1

**Comment by alexforster** [comment]
Hm, that's unfortunate, but it certainly won't be the first API/ABI breakage...

> You must specify the exact version of libbpf-sys, and you must not assume that any other version will be API or ABI compatible.

I think the improved ergonomics are worth it. Once libbpf goes 1.0 (soon) I won't have the chance to make these kinds of changes anymore.

**Comment by alexforster** [comment]
Published: https://crates.io/crates/libbpf-sys/0.5.0-2
